docs/DeveloperGuide.md

```diff
@@ -158,6 +158,64 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 ## **Implementation**
 
 This section describes some noteworthy details on how certain features are implemented.
+### Remove Student feature
+#### Implementation
+
+The removal of Student implements the following operation:
+- `EduTrack#removeStudent(Student s)`  — Removes the Student s from List of Students in EduTrack that tracks all students.
+- `Class#removeStudentFromClass(Student s)`  — Removes the Student s from List of Students in Class that tracks all students in Class.
+
+These operations are exposed in the `Model` interface as `Model#deleteStudent(Student s)` and `Model#deleteStudentFromClass(Student s, Class sClass)`.
+
+Given below is a list of variables used in walkthrough of the removal of Student mechanism.
+
+**Variables used:**
+
+`s` - `Student` to be removed.
+
+`sName` - `Name` representing `s`.
+
+`sClass` - `Class` of `s`.
+
+`studentClassName` - `ClassName` representing `sClass`.
+
+`sClassStudentList` - `List<Student>` in `sClass` containing all its `Student` 
+
+`globalStudentList` - `List<Student>` in `EduTrack` containing all `Student` across all `Class`.
+
+The relationship between variables can be summarised by this object diagram.
+
+<puml src="diagrams/RemoveStudentObjectDiagram.puml" alt="RemoveStudentObjectDiagram" />
+
+**Walkthrough**
+
+Step 1. `LogicManager` calls `removeStudentCommand#execute()` 
+
+Step 1. `removeStudentCommand` calls `Model#getClass(studentClassName)` to get `sClass`.
```

Comment:
Update to DG looks neat and thorough 👍 
There seems to be a repetition of "Step 1", is this by design?